ui/src/components/Bouts/BoutPage.tsx

```diff
@@ -0,0 +1,697 @@
+
+  const [boutStartTime, setBoutStartTime] = useState<Date | undefined>(
+    bout?.startTime && new Date(bout.startTime),
+  );
+  const [boutEndTime, setBoutEndTime] = useState<Date | undefined>(
+    (bout?.endTime && new Date(bout.endTime)) ?? undefined,
+  );
+  const [currentTab, setCurrentTab] = useState(0);
+  const audioCategories = Object.values(AudioCategory);
+  const [audioCategory, setAudioCategory] = useState<AudioCategory | undefined>(
+    targetAudioCategory ?? bout?.category,
+  );
+
+  const timeBuffer = 5; // minutes
+  // Snap to nearest 5 minutes
+  const nearestMinutes = 5;
+  const [timelineStartTime, setTimelineStartTime] = useState<Date>(
+    roundToNearest(
+      subMinutes(targetTime, timeBuffer),
+      nearestMinutes * 60 * 1000,
+      "floor",
+    ),
+  );
+  const [timelineEndTime, setTimelineEndTime] = useState<Date>(
+    min([
+      now,
+      roundToNearest(
+        max([targetTime, addMinutes(targetTime, timeBuffer)]),
+        nearestMinutes * 60 * 1000,
+        "ceil",
+      ),
+    ]),
+  );
+
+  const expandTimelineStart = useCallback(() => {
+    setTimelineStartTime((timelineStartTime) =>
+      roundToNearest(
+        subMinutes(timelineStartTime, timeBuffer),
+        nearestMinutes,
+        "floor",
+      ),
+    );
+  }, [nearestMinutes]);
+
+  const expandTimelineEnd = useCallback(
+    (currentTime: Date) => {
+      setTimelineEndTime((timelineEndTime) =>
+        min([
+          currentTime,
+          roundToNearest(
+            addMinutes(timelineEndTime, timeBuffer),
+            nearestMinutes,
+            "ceil",
+          ),
+        ]),
+      );
+    },
+    [nearestMinutes],
+  );
+
+  const timelineStartTimeMinusADay = subDays(timelineStartTime, 1);
+  // If feed is present, and there's no pre-set time,
+  // get latest stream and last <timeBuffer> minutes of segments.
+  // Set time to end of last segment
+  const feedStreamQueryResult = useListFeedStreamsQuery(
+    {
+      feedId: feed?.id,
+      fromDateTime: timelineStartTime,
+      toDateTime: timelineEndTime,
+      dayBeforeFromDateTime: timelineStartTimeMinusADay,
+    },
+    { enabled: !!feed?.id },
+  );
+
+  // Get detections at current time plus or minus 1 hour
+  const minDetectionsTime = subMinutes(targetTime, 60);
+  const maxDetectionsTime = addMinutes(targetTime, 60);
+  const detectionQueryResult = useDetectionsQuery(
+    {
+      feedId: feed?.id,
+      filter: {
+        timestamp: {
+          greaterThanOrEqual: minDetectionsTime,
+          lessThanOrEqual: maxDetectionsTime,
+        },
+      },
+    },
+    { enabled: !!feed?.id },
+  );
+
+  const feedStreams = useMemo(
+    () => feedStreamQueryResult.data?.feedStreams?.results ?? [],
+    [feedStreamQueryResult],
+  );
+  const feedStream = feedStreams[0];
+  const feedSegments = useMemo(
+    () => feedStreams.flatMap(({ feedSegments }) => feedSegments),
+    [feedStreams],
+  );
```

Comment:
Does this need useMemo?


```suggestion
  const feedStreams = feedStreamQueryResult.data?.feedStreams?.results ?? [];
  const feedStream = feedStreams[0];
  const feedSegments = feedStreams.flatMap(({ feedSegments }) => feedSegments);
```